xc/device.ml

```diff
@@ -1866,6 +1847,32 @@ module Backend = struct
 
       let with_dirty_log domid ~f = f()
 
+      let cmdline_of_info ~xs ~dm info restore domid =
+        let common = Dm_Common.cmdline_of_info ~xs ~dm info restore domid in
+        (* Sort the VIF devices by devid *)
+        let nics = List.stable_sort (fun (_,_,a) (_,_,b) -> compare a b) info.Dm_Common.nics in
+        if List.length nics > Dm_Common.max_emulated_nics then debug "Limiting the number of emulated NICs to %d" Dm_Common.max_emulated_nics;
+        (* Take the first 'max_emulated_nics' elements from the list. *)
+        let rec take n xs = match n, xs with
+          | 0, _ -> []
+          | _, [] -> [] (* return a short list *)
+          | n, x :: xs -> x :: (take (n - 1) xs) in
+        let nics = take Dm_Common.max_emulated_nics nics in
```

Comment:
You can remove `take` and use `Stdext.Listext.List.take`

Reply:
Please try to use `Xapi_stdext_std.Listext` if possilble (And get rid of global Stdext opens when you touch Stdext-related code)